tsdb/chunkenc/histo.go

```diff
@@ -387,14 +386,16 @@ func (a *histoAppender) recode(posInterjections, negInterjections []interjection
 
 	// adopt the new appender into ourselves
 	// we skip porting some fields like schema, t, cnt and zcnt, sum because they didn't change between our old chunk and the recoded one
-	app2 := app.(*histoAppender)
+	app2 := app.(*HistoAppender)
 	a.b = app2.b
 	a.posSpans, a.negSpans = posSpans, negSpans
 	a.posbuckets, a.negbuckets = app2.posbuckets, app2.negbuckets
 	a.posbucketsDelta, a.negbucketsDelta = app2.posbucketsDelta, app2.negbucketsDelta
+
+	return hc, app2
```

Comment:
Here you return the newly created appender, but the caller (I guess) is supposed to just go on with the old appender (the receiver of the method), into which you have copied all the fields.

Reply:
So perhaps don't return app2 to avoid the confusion?

Reply:
no the caller updates the chunk and the appender in head.go, with this code:
```
		} else if len(posInterjections) > 0 || len(negInterjections) > 0 {
			// new buckets have appeared. we need to recode all prior histograms within the chunk before we can process this one.
			s.headChunk.chunk, s.app = app.Recode(posInterjections, negInterjections, sh.PositiveSpans, sh.NegativeSpans)
		}
```

this actually makes it pointless to copy all those fields back into the old appender, so i took that out.